dll/win32/comctl32/static.c

```diff
@@ -626,7 +626,15 @@ static void STATIC_PaintTextfn( HWND hwnd, HDC hdc, DWORD style )
     }
     else
     {
-        DrawTextW( hdc, text, -1, &rc, format );
+        UINT flags = DST_TEXT | ((style & WS_DISABLED) ? DSS_DISABLED : DSS_NORMAL);
+        if (format & DT_HIDEPREFIX)
+            flags |= DSS_HIDEPREFIX;
+        hBrush = GetSysColorBrush(COLOR_GRAYTEXT);
+        DrawStateW(hdc, hBrush, NULL,
```

Comment:
Needs the following callback to be placed _**BEFORE**_ the `STATIC_PaintTextfn()` function:
```c
/**********************************************************************
 *       STATIC_DrawTextCallback
 *
 *   Callback function used by DrawStateW function.
 */
static BOOL CALLBACK STATIC_DrawTextCallback(HDC hdc, LPARAM lp, WPARAM wp, int cx, int cy)
{
    RECT rc;

    SetRect(&rc, 0, 0, cx, cy);
    DrawTextW(hdc, (LPCWSTR)lp, -1, &rc, (UINT)wp);
    return TRUE;
}
```

As you have observed on Windows a non-NULL callback is provided. And this is needed in order to make it use the `format` flags.